src/main/java/net/mcreator/ui/action/accelerators/JAcceleratorButton.java

```diff
@@ -0,0 +1,110 @@
+/*
+ * MCreator (https://mcreator.net/)
+ * Copyright (C) 2012-2020, Pylo
+ * Copyright (C) 2020-2022, Pylo, opensource contributors
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+package net.mcreator.ui.action.accelerators;
+
+import net.mcreator.preferences.PreferencesManager;
+import net.mcreator.themes.ThemeLoader;
+import net.mcreator.ui.action.BasicAction;
+import net.mcreator.ui.dialogs.AcceleratorDialog;
+
+import javax.swing.*;
+import java.awt.*;
+import java.awt.event.*;
+
+/**
+ * <p>This component is a special {@link JButton} used to listen the keyboard and get the new modifiers and the new key code for an {@link Accelerator}.
+ * Some visual aspects are added to help the user.</p>
+ */
+public class JAcceleratorButton extends JButton {

```

Comment:
TMM, this class should be in `net.mcreator.ui.component`, just like there are `JModElementProgressPanel` and `UnsupportedComponent` used by other packages.

Reply:
This class is specific to accelerators, so I decided to keep it in its own package, the same way Validation components are inside `ui.validation.component` https://github.com/MCreator/MCreator/tree/master/src/main/java/net/mcreator/ui/validation/component